### DISCUSSION TITLE
Should touched value update when using `useFieldArray` `update`?

### DISCUSSION THREAD

**Discussion by wootencl** [react-hook-form/react-hook-form#8082] [Q&A] [answered]
Really enjoying this library so much appreciated to the folks/community around it. Did run into a scenario and am not sure entirely of the path forward. Currently I'm using a `useFieldArray` for when of my form elements. The user selects an option from a dropdown which could potentially mutate/update a prior selected item. With that in mind, I'm using the `update` method from `useFieldArray`. Everything seems to be working roughly as expected but the one oddity I've seen is that the `touchedFields` property of `formState` doesn't seem to update _after_ the field has been updated via `update`. The field is marked in `dirtyFields` which seems a bit strange as one would think if a field was dirty it would be touched inherently. Just curious if there's a suggested path forward or possibly if I need to file a bug report. Thanks!

**Comment by Moshyfawn** [comment, marked as answer]
The field **is dirty** if it's **default value is different from it's current one**. The field **is touched** if the user **has interacted with an actual input element**.

You can update the field's value with the `update` field array method, which doesn't mean that you've touched it.

**Reply by wootencl** [reply]
Ah, makes sense I think. Could use `update` strictly programmatically which wouldn't be considered "touching" the field. It does seem like there could possibly be area to expand the `update` API a bit along the lines of `setValue`: `update(index, data, { shouldTouch: true })` 🤔

**Reply by Moshyfawn** [reply]
> to expand the update API a bit along the lines of setValue: update(index, data, { shouldTouch: true }) 🤔

@bluebill1049, what do you think about having the `shouldTouch` as an option to field array `update` method?

**Reply by bluebill1049** [reply]
yea we can, let's have a feature request for that and wait for upvotes. May need to think about all API consistency here for the field array methods as well.

**Reply by Moshyfawn** [reply]
@wootencl, would you mind creating a feature request to support `shouldTouch` with the `update` field array method?
> @bluebill1049: May need to think about all API consistency here for the field array methods as well.

Yeah, makes sense. There're some discrepancies between the single fields and field array API ATM
